grasshopper-loader.js

```diff
@@ -49,26 +51,55 @@ if(require.main === module){
     .option('--type <type>', 'Elasticsearch type within the provided or default index. Defaults to point.', 'point')
     .option('--profile <profile>', 'The aws credentials profile in ~/.aws/credentials. Will also respect AWS keys as environment variables.', 'default')
     .option('--source-srs <sourceSrs>', 'Source Spatial Reference System, passed to ogr2ogr as -s_srs. Auto-detects by default.')
-    .option('--preformatted', 'Input has been preformatted to GeoJson and transformed to WGS84 by ogr2ogr. Results in the loader skipping ogr2ogr and immediately splitting the input into records.')
+    .option('--preformatted', 'Input has been preformatted to GeoJson or csv and transformed to WGS84 by ogr2ogr. Results in the loader skipping ogr2ogr and immediately splitting the input into records.')
     .parse(process.argv);
 
     run(program);
 }
 
-function run(program, loaderCallback){
-  if(!loaderCallback){
-    loaderCallback = function(err){
-      if(err) throw err;
+function run(program, passedCallback){
+
+  if(!passedCallback){
+    passedCallback = function(err){
+      if(err){
+        if(typeof err === "object" && !(err instanceof Error)){
+          err = new Error(JSON.stringify(err));
```

Comment:
If a module calls the errback with {someInfo: 'stuff-you-should-know', processed: 25534, error: err} rather than a plain error.

In fact `verify.js` has a similar signature.. which in retrospect seems a bit odd (idiomatically it's typically `function(err, data)`.. so why I chose to callback with a single object is beyond me.. maybe in an attempt to simplify the function signature.. but then I likely should have still done `function(err, theObject)`). Anywho, I'll be changing `verify.js` as part of the work on loader that brings it to parity with the new path of data in retriever, so I'll fix this then. Nice catch.